BullsAndCowsRunner/Program.cs

```diff
@@ -9,14 +10,42 @@ public static void Main(string[] args)
         {
             SecretGenerator secretGenerator = new SecretGenerator();
             BullsAndCowsGame game = new BullsAndCowsGame(secretGenerator);
+
+            const int maxCount = 6;
+            var count = 0;
             while (game.CanContinue)
             {
                 var input = Console.ReadLine();
+
+                while (!CheckInput(ref input))
+                {
+                    Console.WriteLine("invalid input");
+                    input = Console.ReadLine();
+                }
+
                 var output = game.Guess(input);
                 Console.WriteLine(output);
+                count++;
+
+                if (output == "4A0B" || count == maxCount)
+                {
+                    break;
+                }
             }
 
             Console.WriteLine("Game Over");
         }
+
+        private static bool CheckInput(ref string input)
```

Comment:
This function is better included in the game class?

Reply:
The inputCheck can not block invalid input like "a 1 2 4"

Reply:
good review, check for "a,b,c" is not included